mm2src/coins/tendermint/tendermint_coin.rs

```diff
@@ -1357,6 +1389,29 @@ impl TendermintCoin {
     }
 }
 
+fn find_client(rpc_urls: &[String]) -> Result<(HttpClient, usize), String> {
+    let mut clients = Vec::new();
+    let mut errors = Vec::new();
+    // check that all urls are valid
+    // keep all invalid urls in one vector to show all of them in error
+    for url in rpc_urls.iter() {
+        match HttpClient::new(url.as_str()) {
+            Ok(client) => clients.push(client),
+            Err(e) => errors.push(format!("Url {} is invalid, got error {}", url, e)),
+        }
+    }
+    if !errors.is_empty() {
+        let errors: String = errors.iter().map(|e| format!("{:?}", e)).collect();
+        return Err(errors);
+    }
+    let current = clients.len() - 1;
+    if let Some(client) = clients.pop() {
+        Ok((client, current))
+    } else {
+        Err("None client was found".to_string())
+    }
+}
+
```

Comment:
How about returning a `Vec<HttpClient>` from this function and renaming it to `clients_from_urls`
```rust
fn clients_from_urls(rpc_urls: &[String]) -> Result<Vec<HttpClient>, String> {
    let mut clients = Vec::new();
    let mut errors = Vec::new();
    // check that all urls are valid
    // keep all invalid urls in one vector to show all of them in error
    for url in rpc_urls.iter() {
        match HttpClient::new(url.as_str()) {
            Ok(client) => clients.push(client),
            Err(e) => errors.push(format!("Url {} is invalid, got error {}", url, e)),
        }
    }
    if !errors.is_empty() {
        let errors: String = errors.iter().map(|e| format!("{:?}", e)).collect();
        return Err(errors);
    }
    Ok(clients)
}
```
It will require changes to `TendermintRpcClientImpl` to be
```rust
struct TendermintRpcClientImpl {
    rpc_clients: Vec<HttpClient>,
}
```
While considering the first client in the `vec` the current client. `get_live_client` will need some refactoring too and using `rotate_right` to rotate clients to make the found live client the first one / current.

Reply:
Well, when I made changes, I was guided by the recommendations from this [issue](https://github.com/KomodoPlatform/atomicDEX-API/issues/1480).
I thought we should avoid keeping all clients in vector, but I think rotation will help us to avoid extra iterations over clients and we wont have to keep all rpc urls, client and its index in structure, just have one vector of clietns. I will implement it.

Anyway, if Onur insists on urls vector and one client, we can discuss it again in next PR with z_rpc and web3 refactoring :D

Reply:
Done.
rotation looks cleaner than implementation with `rpc_urls` and `current index`, bcz we don't have to try to create clients again from urls in `get_live_client` func.
Thanks for the idea!

Reply:
> Well, when I made changes, I was guided by the recommendations from this https://github.com/KomodoPlatform/atomicDEX-API/issues/1480.
> Anyway, if Onur insists on urls vector and one client, we will discuss it again in next PR with z_rpc and web3 refactoring :D

Sure, But I think he was just giving rough guidelines in the issue.

> rotation looks cleaner than implementation with rpc_urls and current index, bcz we don't have to try to create clients again from urls in get_live_client func.
Thanks for the idea!

You are always welcome :)